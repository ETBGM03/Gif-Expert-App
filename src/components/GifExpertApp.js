import React, { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [category, setCategory] = useState(defaultCategories);

  return (
    <div>
      <h2>Gif Expert App</h2>
      <AddCategory setCategory={setCategory} />
      <hr />
      <ol>
        {category.map((categ) => {
          return <GifGrid key={categ} category={categ} />;
        })}
      </ol>
    </div>
  );
};

export default GifExpertApp;
