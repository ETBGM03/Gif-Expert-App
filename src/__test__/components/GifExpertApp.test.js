import { shallow } from "enzyme";
import React from "react";
import GifExpertApp from "../../components/GifExpertApp";

describe("probando el componente <GifExpertApp />", () => {
  test("debe mostrarse correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const catageories = ["Colombia", "pereira"];
    const wrapper = shallow(<GifExpertApp defaultCategories={catageories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(catageories.length);
  });
});
