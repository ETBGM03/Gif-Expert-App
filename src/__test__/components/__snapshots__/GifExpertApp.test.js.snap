// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Colombia"
      key="Colombia"
    />
    <GifGrid
      category="pereira"
      key="pereira"
    />
  </ol>
</div>
`;

exports[`probando el componente <GifExpertApp /> debe mostrarse correctamente 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</div>
`;
